packages/fetch/src/fetch.client.ts

```diff
@@ -33,16 +45,17 @@ export class FetchClient<Opts extends FetchOptions<any, any> = RequestInit, Req
     const { json, ...options } = rawOptions;
 
     const headers = new Map(options.headers && FetchClient.extractHeaders(options.headers));
+    const searchParams = new URLSearchParams(options.searchParams && FetchClient.extractSearchParams(options.searchParams));
     let { body } = options;
 
     if (json != null) {
       headers.set('content-type', 'application/json');
       body = JSON.stringify(json);
     }
 
-    const finalURL = typeof url === 'string' ? `${this.config.baseURL ?? ''}${url}` : url;
+    const finalURL = typeof url === 'string' ? FetchClient.formatURL(this.config.baseURL, url, searchParams) : url;
     const response = await this.raw(finalURL, {
-      ...options,
```

Comment:
I think we need to still spread `options` here otherwise we won't pass all the request options through to the `fetch` call

Reply:
Right. I will extract out the options that we're modifying then before this spread.